packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts

```diff
@@ -210,16 +216,13 @@ class MessageEventBus extends EventEmitter {
 		// this is for internal use ONLY and not for use with custom destinations!
 		this.emit('message', msg);
 
-		LoggerProxy.debug(`Listeners: ${this.eventNames().join(',')}`);
+		// LoggerProxy.debug(`Listeners: ${this.eventNames().join(',')}`);
 
-		// if there are no set up destinations, immediately mark the event as sent
 		if (
-			!isLogStreamingEnabled() ||
-			Object.keys(this.destinations).length === 0 ||
-			!this.hasAnyDestinationSubscribedToEvent(msg)
+			isLogStreamingEnabled() &&
+			Object.keys(this.destinations).length > 0 &&
+			this.hasAnyDestinationSubscribedToEvent(msg)
```

Comment:
we have this almost identical check in `send`. we could extract this into a private function 
```typescript
	private shouldSend(msg: EventMessageTypes): boolean {
		return (
			isLogStreamingEnabled() &&
			Object.keys(this.destinations).length > 0 &&
			this.hasAnyDestinationSubscribedToEvent(msg)
		);
	}
```

and update this to 
```suggestion
		shouldSend(msg)
```